### PR TITLE
Add the ability to mock BatchGet and MultiTableBatchGet operations

### PR DESCRIPTION
**Pull request by 96malhar**
## Description
This PR adds the ability to mock `BatchGet` and `MultiTableBatchGet` operations in DynamoDB.

* The `IDynamoDBContext.CreateBatchGet<T>(...)` method now returns a generic `IBatchGet<T>` interface
* The `IDynamoDBContext.CreateMultiTableBatchGet(...)` method now returns a `IMultiTableBatchGet` interface.

The `IBatchGet<T>` interface also implements the non generic `IBatchGet` which defines the `Execute` and `ExecuteAsync` method to carry out the BatchGet operations.

I have added new unit tests to showcase how a user will mock these interfaces.

## Motivation and Context

DOTNET-7676

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [X] I have read the **README** document
- [X] I have added tests to cover my changes
- [X] All new and existing tests passed

## License

- [X] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs [hunk 1]
```diff
@@ -27,318 +25,295 @@
...
+        /// </returns>
+        IMultiTableBatchGet Combine(params IBatchGet[] otherBatches);
+    }
 
-        #region Internal methods
+    /// <summary>
+    /// Represents a non-generic object for retrieving a batch of items
+    /// from a single DynamoDB table
+    /// </summary>
+    public abstract class BatchGet
```
Comment (normj) [inline sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs:104]: This should be implementing `IBatchGet`.

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs [hunk 2]
```diff
@@ -27,318 +25,295 @@
...
-        /// List of results retrieved from DynamoDB.
-        /// Populated after Execute is called.
-        /// </summary>
-        new public List<T> Results { get { return TypedResults; } }
+        /// <inheritdoc/>
+        public int TotalKeys => _keys.Count;
 
-        #endregion
+        /// <inheritdoc/>
+        public List<object> UntypedResults { get; private set; } = new List<object>();
```
Comment (normj) [inline sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs:128]: This property should be on the base class `BatchGet` that implements `IBatchGet` the non-generic versions.

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs [hunk 3]
```diff
@@ -27,318 +25,295 @@
...
 
+        /// <inheritdoc/>
+        public List<T> Results { get; private set; } = new List<T>();
 
-        #region Public methods
+        /// <inheritdoc/>
+        public bool ConsistentRead { get; set; }
```
Comment (normj) [inline sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs:134]: This property should be on the base class `BatchGet` that implements `IBatchGet` the non-generic versions.

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs (hunk 2 above)
Comment (96malhar) [reply sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs:128]: Done

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs (hunk 3 above)
Comment (96malhar) [reply sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs:134]: Done

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs (hunk 1 above)
Comment (96malhar) [reply sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs:104]: Done

File: sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContext.cs [hunk 4]
```diff
@@ -113,22 +112,23 @@ public partial interface IDynamoDBContext : IDisposable
...
+
         /// <summary>
         /// Creates a strongly-typed BatchGet object, allowing
         /// a batch-get operation against DynamoDB.
         /// </summary>
         /// <typeparam name="T">Type of objects to get</typeparam>
         /// <param name="operationConfig">Config object which can be used to override that table used.</param>
-        /// <returns>Empty strongly-typed BatchGet object</returns>
-        BatchGet<T> CreateBatchGet<T>(DynamoDBOperationConfig operationConfig = null);
+        /// <returns>An interface with the ability to perform BatchGet operations</returns>
```
[the remark this answers is not in the document]
Comment (96malhar) [reply sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContext.cs:122]: Done

**Review by normj — APPROVED** [review]

**Review by normj — APPROVED** [review]

**Review by ashovlin — APPROVED** [review]

**Pull request merged**